modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageWriteHandler.java

```diff
@@ -66,6 +76,11 @@ class MetaStorageWriteHandler {
     boolean handleWriteCommand(CommandClosure<WriteCommand> clo) {
         WriteCommand command = clo.command();
 
+        if (command instanceof MetaStorageWriteCommand) {
```

Comment:
No-no-no, please update timestamp _after_ you finished processing :)
Otherwise you'll have data races